### PR TITLE
Escape special characters

### PR DESCRIPTION
**Pull request by aaronjwood**
This should help with #16580. I ignored a few instances I came across in the test files.

### CODE REVIEW

File: apps/user_ldap/lib/Mapping/AbstractMapping.php [hunk 1]
```diff
@@ -147,7 +147,7 @@ public function getNamesBySearch($search) {
 			WHERE `owncloud_name` LIKE ?
 		');
 
-		$res = $query->execute(array($search));
+		$res = $query->execute(array($this->connection->escapeLikeParameter($search)));
```
Comment (aaronjwood) [inline apps/user_ldap/lib/Mapping/AbstractMapping.php]: I'm not sure if I did the right thing here. Can someone confirm? Should `$search` be able to contain characters such as `_` and/or `%`?

**Comment by aaronjwood** [comment]
I state that my contributions here are MIT licensed.

**Comment by PVince81** [comment]
- [x] TEST: custom webdav prop can be set with PROPPATCH and retrieved with PROPFIND
- [ ] TEST: LDAP with users and groups still works
- legacy storage stuff is obsolete and will only run when upgrading from 8.0 (but good to have the fix there still)

:+1:

**Comment by PVince81** [comment]
Jenkins PR so we can get CI results: https://github.com/owncloud/core/pull/25433

**Comment by DeepDiver1975** [comment]
Some tests are failing

````
05:14:21 There were 3 failures:
05:14:21 
05:14:21 1) OCA\DAV\Tests\unit\Connector\Sabre\CustomPropertiesBackendTest::testGetPropertiesForDirectory
05:14:21 Failed asserting that null matches expected 'value1'.
05:14:21 
05:14:21 /ssd/jenkins/workspace/core-ci-linux-php5.4/database/sqlite/label/SLAVE/apps/dav/tests/unit/Connector/Sabre/CustomPropertiesBackendTest.php:281
05:14:21 
05:14:21 2) OCA\User_LDAP\Tests\Mapping\GroupMappingTest::testSearch
05:14:21 Failed asserting that 0 is identical to 2.
05:14:21 
05:14:21 /ssd/jenkins/workspace/core-ci-linux-php5.4/database/sqlite/label/SLAVE/apps/user_ldap/tests/Mapping/AbstractMappingTest.php:169
05:14:21 
05:14:21 3) OCA\User_LDAP\Tests\Mapping\UserMappingTest::testSearch
05:14:21 Failed asserting that 0 is identical to 2.
05:14:21 
05:14:21 /ssd/jenkins/workspace/core-ci-linux-php5.4/database/sqlite/label/SLAVE/apps/user_ldap/tests/Mapping/AbstractMappingTest.php:169
05:14:21

**Comment by aaronjwood** [comment]
@DeepDiver1975 thanks for the heads up. I've added a few more fixes that I believe should resolve the failures. Let me know your thoughts on my changes since I've changed the signature of `getNamesBySearch`

**Comment by PVince81** [comment]
Thanks, I have pushed the changes to the Jenkins branch: https://github.com/owncloud/core/pull/25433

**Comment by PVince81** [comment]
Tests passed apparently.

@DeepDiver1975 merge ?

**Comment by PVince81** [comment]
stable9.1: https://github.com/owncloud/core/issues/26340
stable9: https://github.com/owncloud/core/issues/26409
stable8.2: https://github.com/owncloud/core/issues/26410

**Pull request merged**